sdk/clientcore/core/src/main/java/io/clientcore/core/models/binarydata/BinaryData.java

```diff
@@ -681,10 +718,14 @@ public <T> T toObject(Type type) throws IOException {
      *
      * @param outputStream The {@link OutputStream} to write the contents of this {@link BinaryData} to.
      * @throws NullPointerException If {@code outputStream} is null.
-     * @throws IOException If an I/O error occurs.
+     * @throws ClientCoreException If an error occurs during reading the content or writing it to the stream.
```

Comment:
this is the example when we don't document `UncheckedIOException`  (potentially coming from `toBytes()`)

I.e. users need to handle both - `IOException` and, if they discover it, `UncheckedIOException`. 
I think it's more friendly to throw one exception for both of them.